nrpe_exporter.go

```diff
@@ -60,12 +62,29 @@ func collectCommandMetrics(cmd string, conn net.Conn, logger log.Logger) (Comman
 
 // Collect dials nrpe-server and issues given command, recording metrics based on the result.
 func (c *Collector) Collect(ch chan<- prometheus.Metric) {
+	var ctx *openssl.Ctx
+	var conn net.Conn
+	var err error
 	// Connect to NRPE server
-	conn, err := net.Dial("tcp", c.target)
+	if c.ssl {
+		ctx, err = openssl.NewCtx()
+		if err != nil {
+			goto return_err
+		}
+		err = ctx.SetCipherList("ALL:!MD5:@STRENGTH")
+		if err != nil {
+			goto return_err
+		}
+		conn, err = openssl.Dial("tcp", c.target, ctx, openssl.InsecureSkipHostVerification)
+	} else {
+		conn, err = net.Dial("tcp", c.target)
+	}
+return_err:
```

Comment:
I'd prefer to duplicate rather than goto.

Reply:
Done